docs/sources/reference/components/prometheus.exporter.snowflake.md

```diff
@@ -20,18 +22,32 @@ prometheus.exporter.snowflake "LABEL" {
 }
 ```
 
+### RSA Authentication
+
+```alloy
+prometheus.exporter.snowflake "LABEL" {
+    account_name =         ACCOUNT_NAME
+    username =             USERNAME
+    private_key_path =     /PATH/TO/rsa_key.p8
+    private_key_password = PASSWORD
+    warehouse =            WAREHOUSE
+}
+```
+
 ## Arguments
 
 The following arguments can be used to configure the exporter's behavior.
 Omitted fields take their default values.
 
-| Name           | Type     | Description                                           | Default          | Required |
-| -------------- | -------- | ----------------------------------------------------- | ---------------- | -------- |
-| `account_name` | `string` | The account to collect metrics for.                   |                  | yes      |
-| `username`     | `string` | The username for the user used when querying metrics. |                  | yes      |
-| `password`     | `secret` | The password for the user used when querying metrics. |                  | yes      |
-| `role`         | `string` | The role to use when querying metrics.                | `"ACCOUNTADMIN"` | no       |
-| `warehouse`    | `string` | The warehouse to use when querying metrics.           |                  | yes      |
+| Name                   | Type     | Description                                                                                  | Default          | Required |
+| ---------------------- | -------- | -------------------------------------------------------------------------------------------- | ---------------- | -------- |
+| `account_name`         | `string` | The account to collect metrics for.                                                          |                  | yes      |
+| `username`             | `string` | The username for the user used when querying metrics.                                        |                  | yes      |
```

Comment:
username, password, private_key_path and private_key_password should all be optional from not on, no?

Reply:
Additionally, we should add some documentation explaining how to use these variables. Like, one or the other, precedence order, etc.

Reply:
Agreed on the first point, but there are scenarios where these are required, so some thought this qualified as 'required'. I had an earlier version with more details outside of the table, but opted to add to the field descriptions instead. I'll add these comments back in, let me know what you think.